euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/Context.java

```diff
@@ -16,27 +16,8 @@
 package cz.seznam.euphoria.core.client.io;
 
 /**
- * A collector of elements. Used in functors.
- *
- * @param <T> the type of elements collected through this context
+ * Used in user defined functions to access environment methods.
  */
-public interface Context<T> {
-
-  /**
-   * Collects the given element to the output of this context.
-   *
-   * @param elem the element to collect
-   */
-  void collect(T elem);
-
-  /**
-   * Retrieves the window - if any - underlying the current
-   * execution of this context.
-   *
-   * @return {@code null} if this context is not executed within a
-   *          windowing strategy, otherwise the current window of
-   *          this context
-   */
-  Object getWindow();
+public interface Context extends Environment {
```

Comment:
Apart from the changes introduce by @vanekjar, there's troubles in general with `FlatMap` in regards to having to specify the types of the lambdas params. The generic parameter of the old "context" object (now it's called `Collector`) of the UDF is not bound by any preceding expression, and, thus, the user has to explicitly specify the types of the parameters explicitly.

If you check the usage of the `.using` method on `master` you'll see the pattern. That's why the change by Jara isn't that dramatic after all.